lib/standard/string.nit

```diff
@@ -895,6 +902,64 @@ abstract class FlatText
 	redef fun flatten do return self
 end
 
+# Used to keep track of the last accessed char in a String
+class CharCache
+	# The position (as in char) of a String
+	var position: Int
+	# The position in the NativeString underlying the String
+	var bytepos: Int
+end
+
+redef class UnicodeChar
+
+	protected fun to_ns(ns: NativeString, len: Int) is extern "u8char_to_ns"
+
+	redef fun to_s do
+		var ln = len
+		var ns = new NativeString(len + 1)
+		to_ns(ns, len)
+		return ns.to_s
+	end
+
+	redef fun output is extern "u8char_output"
+
+	# Real length of the char in UTF8
+	#
+	# As per the specification :
+	#
+	# ~~~raw
+	#  Length  |        UTF-8 octet sequence                     |  Min  |  Max
+	#          |              (binary)                           | Value | Value
+	# ---------+-------------------------------------------------+-------+---------
+	#  1       | 0xxxxxxx                                        |   0   |  127
+	#  2       | 110xxxxx 10xxxxxx                               |  128  |  2047
+	#  3       | 1110xxxx 10xxxxxx 10xxxxxx                      | 2048  |  65535
+	#  4       | 11110xxx 10xxxxxx 10xxxxxx 10xxxxxx             | 65536 |  2097151
+	# ~~~
+	private fun len: Int is extern "u8char_length"
+
+	# Returns the Unicode code point representing the character
+	#
+	# Note : A unicode character might not be a visible glyph, but it will be used to determine canonical equivalence
```

Comment:
line overflow, cut it :)

it is possible to add some nitunits? (also in the other methods of the class)